### PR TITLE
Scene pipeline documentation

### PR DESCRIPTION
**Pull request by jackalbe**
### Submission Checklist:

* [ ] **Descriptive active voice** - Do descriptive sentences have a clear *subject* and *action verb*?
* [ ] **Answer the question at hand** - Does the documentation answer a *what*, *why*, *how*, or *where* type of question?
* [ ] **Consistency** - Does the content consistently follow the [style guide](https://o3de.org/docs/contributing/to-docs/style-guide/)?
* [ ] **Help the user** - Does the documentation show the user something *meaningful*?

### CODE REVIEW

File: content/docs/user-guide/assets/scene-pipeline/scene-graph.md [hunk 1]
```diff
@@ -0,0 +1,170 @@
...
+{{< /note >}}
+
+### Guaranteed root node
+
+The ```SceneGraph``` is guaranteed to have at least one node. By default, this is a nameless node at the root. This guarantee avoids additional complexities having to deal with the edge case of no nodes. This both helps reduce the API for the ```SceneGraph``` and makes it easier to work with as there's no need for any tests to deal with having no entries. It is something though that has to be kept in mind if a blank root node is not an option. This root node behaves in the same way as other nodes so data and siblings can be assigned to it.
+
+## Internal layout
+
+{{< note >}}
+ADD GRAPH
```
Comment (AMZN-stankowi) [inline content/docs/user-guide/assets/scene-pipeline/scene-graph.md:38]: This shouldn't get checked in. I think bringing in these docs changes in two steps, 1 without images and a second pull request that adds images will be fine.

**Review by AMZN-stankowi — COMMENTED** [review]
The image scene-pipe-scene-setttings.png should be updated, it's showing Lumberyard in the title bar instead of O3DE. It also should be updated with the latest visuals of this tool.

File: content/docs/user-guide/assets/pipeline/user_defined_properties.md [hunk 2]
```diff
@@ -113,8 +87,38 @@ def print_properties(scene):
...
 ![Car FBX example with UDPs.](/images/user-guide/assets/pipeline/scene_api_udp.jpg)
+
+## Future Use Cases of UDP metadata
+
+There are many use cases for UDP metadata in both the O3DE engine and for content teams in the future. For any soft naming rule such as naming nodes with a ```_lod0``` to indicate level of detail (LOD), a user defined property key can be used instead.
+
+These are a few features that will be managed using user defined properties in the future.
+
+{{< note >}}
+None of these UDP metadata attributes are active in the code base as of August 2022.
```
[the remark this answers is not in the document]
Comment (jackalbe) [reply content/docs/user-guide/assets/pipeline/user_defined_properties.md:103]: removed

File: content/docs/user-guide/assets/scene-pipeline/scene-graph.md (hunk 1 above)
Comment (jackalbe) [reply content/docs/user-guide/assets/scene-pipeline/scene-graph.md]: added

File: content/docs/user-guide/assets/scene-pipeline/scene-manifest.md [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+
+After loading a source scene there is always a ```SceneManifest``` present. Attempts will first be made to read the ```SceneManifest``` from a file with the extension .assetinfo. This name is appended to the file path of the source file, such that “example/file.fbx” will cause the Scene pipeline to look for the manifest in “example/file.fbx.assetinfo”. If the file is not found the Scene pipeline will create a new manifest in memory that may or may not get saved to disk later. Rather than create a blank manifest or one with a few default values, several smaller pieces of code (see Components – Behaviors) analyze the ```SceneGraph``` and use various pieces of logic and knowledge to construct a manifest that is likely to produce the results the user is looking for. If a manifest is successfully loaded these same pieces of code get a chance to look at the state of the manifest and the graph so any discrepancies such as added/removed.
```
Comment (junbo75) [inline content/docs/user-guide/assets/scene-pipeline/scene-manifest.md:25]: > If a manifest is successfully loaded these same pieces of code get a chance to look at the state of the manifest and the graph so any discrepancies such as added/removed.

This sentence seems incomplete to me. Do you mean something like this?
"If a manifest is successfully loaded these same pieces of code get a chance to look at the state of the manifest, the graph and any discrepancies such as added/removed."

File: content/docs/user-guide/assets/pipeline/user_defined_properties.md [hunk 4]
```diff
@@ -15,33 +15,7 @@ The CustomPropertyData can be accessed by scene builders using either the Python
 
 ## Assigning Materials in Default Procedural Prefab using UDPs
```
[the remark this answers is not in the document]
Comment (jackalbe) [reply content/docs/user-guide/assets/pipeline/user_defined_properties.md:16]: that does make sense

File: content/docs/user-guide/assets/scene-pipeline/scene-api.md [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+
+SceneCore is the central library for the Asset Importer framework. It provides the storage for a graph (```SceneGraph```) that contains the in-memory representation of the imported data and a dictionary (```SceneManifest```) that contains the groups and rules that guide the exporting process. It also contains various utilities to make it easier to work with the graph and manifest. Lastly it contains a collection of interfaces that provide the a common "language" for the import and export process to talk. Some of these interfaces are mandatory, such as the IManifestObject, in order for base functionality to be implemented. Others are optional, but advisable to implement as they make code reusable. As an example, if custom mesh data is loaded, it's advisable to have the data container implement the IMesh interface so the UI knows when to present the mesh to the user as a selectable option and for an exporter to know how to interpret the data. Of course, if this isn't desired, not implementing the IMesh interface will disable this functionality, but the new data can still be made available in the graph.
```
[the remark this answers is not in the document]
Comment (jackalbe) [reply content/docs/user-guide/assets/scene-pipeline/scene-api.md:15]: I am not certain what the change you want here. This is the following sentences: 

> Some of these interfaces are mandatory, such as the IManifestObject, in order for base functionality to be implemented. Others are optional, but advisable to implement as they make code reusable. 

Which 'a' or 'the' do you want removed?

File: content/docs/user-guide/assets/scene-pipeline/scene-builder.md [hunk 6]
```diff
@@ -0,0 +1,175 @@
...
+
+This design would make it possible to do other types of processes to modify the scene graph, like mesh splitting, or dynamic animation generation.
+
+## Creating a scene builder gem
+
+The scene builder framework is used in tool gems called Builder gems. The gem adds classes that derive from the scene building components base classes:  `AZ::SceneAPI::SceneCore::LoadingComponent`, `AZ::SceneAPI::SceneCore::GenerationComponent`, `AZ::SceneAPI::SceneCore::BehaviorComponent`, or `AZ::SceneAPI::SceneCore::ExportingComponent`. The new components use the `BindToCall' method to hook into the desired scene build events. Finally, the class descriptors are added to the gem module description.
```
[the remark this answers is not in the document]
Comment (jackalbe) [reply content/docs/user-guide/assets/scene-pipeline/scene-builder.md:21]: This section seeks to define and explain scene builders, so defining the term again feels like appropriate text. In particular this section seeks to explain what parts of the gem structure needs to be enabled (it is a Tools gem) and what the CMake alias calls it (Builder).

File: content/docs/user-guide/assets/scene-pipeline/scene-api.md (hunk 5 above)
[the remark this answers is not in the document]
Comment (junbo75) [reply content/docs/user-guide/assets/scene-pipeline/scene-api.md:15]: I mean, this sentence actually:

> Lastly it contains a collection of interfaces that provide **the a** common "language" for the import and export process to talk.

Sorry but all the sentences here are on the same line, so it's hard to comment on the exact sentence :)

File: content/docs/user-guide/assets/scene-pipeline/scene-builder.md (hunk 6 above)
[the remark this answers is not in the document]
Comment (junbo75) [reply content/docs/user-guide/assets/scene-pipeline/scene-builder.md:21]: Just a minor thing: In the previous section, it is called "scene builder gem" but we call it "Builder gem" here. Maybe keep them consistent and use "scene builder gem" here as well.

File: content/docs/user-guide/assets/scene-pipeline/scene-graph.md [hunk 7]
```diff
@@ -0,0 +1,168 @@
+---
+linkTitle: Scene Graph
+title: Scene Graph
+description: The scene node contents and hierarchal data from a source scene.
+weight: 400
+toc: true
+---
+
+The ```SceneGraph``` is a bi-directional a-cyclical graph targeting a small number of named entries (maximum of 2 million). It supports both hierarchical and flat iteration over the stored data, either through classic index-based retrieval or through custom iterators. Nodes in the tree can be quickly located through a path-like name that's composited for all entries.
```
[the remark this answers is not in the document]
Comment (jackalbe) [reply content/docs/user-guide/assets/scene-pipeline/scene-graph.md:9]: the graph does not really support cycles... rather references 
so maybe "bi-directional acyclical graph"

File: content/docs/user-guide/assets/scene-pipeline/scene-graph.md [hunk 8]
```diff
@@ -0,0 +1,168 @@
...
+        AZ_TracePrintf("Attribute: %s\n", sceneGraph.GetNodeName(current).c_str());
+    }
+    current = sceneGraph.GetNodeSibling(current);
+}
+```
+
+### Iterator based
+
+Besides index based access, the content of the ```SceneGraph``` can be accessed through iterators which behave much like iterators from common containers such as AZStd::vector or AZStd::unordered_map. To reduce the number of functions to deal with, the begin and end iterator are combined into a "view". Besides reducing the number of functions, views also allow the use of ranged for-loops without the need for additional wrapper classes. For convenience all the iterators that are part of the ```SceneGraph``` provide utility functions in the form of ```Make<iterator>View``` to easily construct views.
+Non-hierarchical
```
[the remark this answers is not in the document]
Comment (jackalbe) [reply content/docs/user-guide/assets/scene-pipeline/scene-graph.md:75]: This section describes how to use the Scene API iterators and view. So maybe "Iterator based traversal" ?

[the remark this answers is not in the document]
Comment (junbo75) [reply content/docs/user-guide/assets/scene-pipeline/scene-graph.md:77]: image
Sorry but I mean the file format doesn't seem quite correctly here in the preview. "Non-hierarchical" is shown as body content but I think it should be a header/title.

File: content/docs/user-guide/assets/scene-pipeline/user_defined_properties.md [hunk 9]
```diff
@@ -15,33 +15,7 @@ The CustomPropertyData can be accessed by scene builders using either the Python
...
-
-### Advanced Custom Component
-
-Presently the default procedural prefab will assemble an entity with a mesh component and an optional material component for each mesh data node if finds in the scene graph. In the future we plan to add a ```o3de.entity.component``` property to take in a string value. This will be used in conjunction with an event bus to allow custom scene builders to add a custom component with the entity as well. This will allow content teams to add editor components that describe custom project data such as hit points or trigger event name or anything that a game team wants to assign with a mesh node from a DCC tool.
+The default procedural prefab scene builder inside the Prefab game has an example of how UDP metadata can be used to customized the scene pipeline. The Prefab gem constructs a default prefab for a scene graph, it looks for the UDP ```o3de_default_material``` in a mesh data node to define an O3DE render material for that mesh group inside the prefab. This allows artists to assign O3DE render materials back in the DCC tool. Each time the source scene file (i.e. FBX) is exported, this reference is maintained through the scene pipeline.
```
Comment (amzn-mike) [inline content/docs/user-guide/assets/scene-pipeline/user_defined_properties.md:18]: Typo: customized -> customize

**Comment by willihay** [comment]
Note: Before merging, this PR should either be retargeted to the stabilization/22.10 branch or cherry-picked after merge to that branch.

**Review by AMZN-stankowi — APPROVED** [review]

**Review by junbo75 — APPROVED** [review]
Looks good to me. There're a few minor typo and formatting issue. Please check the previous unresolved comments on this PR.

File: content/docs/user-guide/assets/_index.md [hunk 10]
```diff
@@ -11,6 +11,7 @@ Topics in this section cover the **Asset Pipeline** and asset processing, includ
 | Topic Area | Description |
 | --- | --- |
 | [Asset Pipeline](pipeline) | The Asset Pipeline is the end-to-end process that transforms source assets into runtime optimized product assets. |
+| [Scene Pipeline](scene-pipeline) | The Scene Pipeline facilitates the conversion from source scene assets that loads, converts, and export assets containing scene engine data.  |
 | [Asset Processor](asset-processor) | Asset Processor discovers source assets, manages asset process jobs, and maintains the **Asset Cache**. |
 | [Scene Settings](scene-settings) | Use Scene Settings to customize how meshes, actors, motions, and PhysX colliders are processed. |
 | **Texture Settings** | Use Texture Settings to customize how source image assets are processed. |
```
Comment (micronAMZN) [inline content/docs/user-guide/assets/_index.md:17]: Seems the branch for this PR is out of date. This texture settings documentation has been added and is in stabilization. Please make sure this branch has merged the latest stabilization/22.10 and resolve any issues/conflicts.

**Review by micronAMZN — CHANGES_REQUESTED** [review]
This branch needs to be updated to the latest stabilization/22.10 and issues/conflicts need to be resolved (if any).
